sklearn/datasets/tests/test_common.py

```diff
@@ -87,6 +87,7 @@ def _generate_func_supporting_param(param, dataset_type=("load", "fetch")):
             yield pytest.param(name, obj, marks=marks)
 
 
+@pytest.mark.xfail
```

Comment:
I think this is better to add an entry in the `KNOWN_FAILURE` on the top of the file:

```python
KNOWN_FAILURE = {
    "return_X_y": {
        "fetch_20newsgroups": pytest.mark.xfail(
            reason="X is a list and does not have a shape argument"
        ),
        "fetch_openml": pytest.mark.xfail(
            reason="fetch_opeml requires a dataset name or id"
        ),
        "fetch_lfw_people": pytest.mark.xfail(
            reason="fetch_lfw_people can fail if pillow is not installed"
        )
    },
    "as_frame": {
        "fetch_openml": pytest.mark.xfail(
            reason="fetch_opeml requires a dataset name or id"
        ),
    }
}
```

Reply:
You are absolutely right @glemaitre!